### PR TITLE
[SRTS-A01](Aqara Smart Radiator Thermostat E1) add external sensor

### PR DESCRIPTION
**Pull request by artem-sedykh**

### CODE REVIEW

File: devices/xiaomi.js [hunk 1]
```diff
@@ -121,10 +139,93 @@ const tzLocal = {
...
+                    await entity.write('aqaraOpple', {0xfff2: {value: val1, type: 0x41}}, {manufacturerCode: 0x115f});
+                    await entity.write('aqaraOpple', {0xfff2: {value: val2, type: 0x41}}, {manufacturerCode: 0x115f});
+
+                    await entity.read('hvacThermostat', ['localTemp']);
+                }
+                break;
+            }
+            case 'sensor_temp':
+                if (meta.state['sensor'] === 'external') {
+                    const temperatureBuf = Buffer.allocUnsafe(4);
```
Comment (Koenkk) [inline devices/xiaomi.js:202]: Why not use `Buffer.alloc`? (https://www.w3schools.com/nodejs/met_buffer_allocunsafe.asp)

File: devices/xiaomi.js [hunk 2]
```diff
@@ -2282,16 +2383,28 @@ module.exports = [
...
+        exposes: [
+            exposes.switch()
+                .withState('state', true, 'Turn off the thermostat', ea.ALL, 'OFF', 'ON'),
+            exposes.climate()
+                .withSetpoint('occupied_heating_setpoint', 5, 30, 0.5)
+                .withLocalTemperature(ea.STATE)
+                .withPreset(['manual', 'away', 'auto']).setAccess('preset', ea.ALL),
+            e.temperature_sensor_select(['internal', 'external']).withAccess(ea.ALL),
+            exposes.binary('calibrated', ea.STATE, true, false)
+                .withDescription('Is the valve calibrated'),
```
Comment (Koenkk) [inline devices/xiaomi.js:2395]: Can you explain a bit more what this does?

Comment (artem-sedykh) [reply devices/xiaomi.js:2395]: "calibrated": the sign of valve calibration has the values 1 - the valve is calibrated and 0 - the valve is not calibrated
"sensor": internal - internal temperature sensor is used; external - an external temperature sensor is used

File: devices/xiaomi.js (hunk 1 above)
Comment (artem-sedykh) [reply devices/xiaomi.js]: the allocUnsafe method works faster than alloc, I'll fix it for alloc

File: devices/xiaomi.js (hunk 2 above)
Comment (Koenkk) [reply devices/xiaomi.js:2392]: But was does "valve calibrated" mean?

Comment (artem-sedykh) [reply devices/xiaomi.js:2392]: Valve finds extreme positions, first it closes completely, and then it opens completely, after that it is considered that the thermal head can control the temperature

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**